client/GameComponents/Messages.jsx

```diff
@@ -24,12 +26,30 @@ class InnerMessages extends React.Component {
     getMessage() {
         var index = 0;
         var messages = _.map(this.props.messages, message => {
-            return <div key={'message'+index++} className='message'>{message.message}</div>;
+            return <div key={'message'+index++} className='message'>{this.formatMessageText(message.message)}</div>;
         });
 
         return messages;
     }
 
+    formatMessageText(message) {
+        var index = 0;
+        return _.map(message, fragment => {
+            if(fragment.code && fragment.label) {
```

Comment:
This check felt hacky but we may be able to wait until we introduce other argument types (e.g. challenge icons) to deal with it.